test/e2e/node/crictl.go

```diff
@@ -38,37 +38,25 @@ var _ = SIGDescribe("crictl", func() {
 	})
 
 	ginkgo.It("should be able to run crictl on the node", func() {
-		// Get all nodes' external IPs.
-		ginkgo.By("Getting all nodes' SSH-able IP addresses")
-		hosts, err := e2essh.NodeSSHHosts(f.ClientSet)
-		if err != nil {
-			framework.Failf("Error getting node hostnames: %v", err)
-		}
-		e2eskipper.SkipUnlessAtLeast(len(hosts), 1, "Skipping test because found no SSH'able node")
+		nodes, err := e2enode.GetBoundedReadySchedulableNodes(f.ClientSet, maxNodes)
+		framework.ExpectNoError(err)
 
-		testCases := []struct {
-			cmd string
-		}{
-			{`sudo crictl version`},
-			{`sudo crictl info`},
+		testCases := []string{
+			"sudo crictl version",
+			"sudo crictl info",
 		}
 
+		hostExec := utils.NewHostExec(f)
+
 		for _, testCase := range testCases {
-			for _, host := range hosts {
-				ginkgo.By(fmt.Sprintf("SSH'ing to node %q to run %q", host, testCase.cmd))
+			for _, node := range nodes.Items {
+				ginkgo.By(fmt.Sprintf("Testing %q on node %q ", testCase, node.GetName()))
 
-				result, err := e2essh.SSH(testCase.cmd, host, framework.TestContext.Provider)
-				stdout, stderr := strings.TrimSpace(result.Stdout), strings.TrimSpace(result.Stderr)
-				if err != nil {
-					framework.Failf("Ran %q on %q, got error %v", testCase.cmd, host, err)
-				}
-				// Log the stdout/stderr output.
-				// TODO: Verify the output.
-				if len(stdout) > 0 {
-					framework.Logf("Got stdout from %q:\n %s\n", host, strings.TrimSpace(stdout))
-				}
-				if len(stderr) > 0 {
-					framework.Logf("Got stderr from %q:\n %s\n", host, strings.TrimSpace(stderr))
+				res, err := hostExec.Execute(testCase, &node)
+				framework.ExpectNoError(err)
+
+				if res.Stdout == "" {
+					framework.Fail("stdout is empty")
```

Comment:
just a question: Does this new string length check solve the original TODO of the original line 66?
I am not sure what was the intention of the original TODO.

Reply:
I think so. I was considering to test the output directly but depending on the `crictl` version it may differ. Getting anything on stdout seems fine from my point of view for now.